cmds/exp/disk_unlock/disk_unlock.go

```diff
@@ -162,20 +191,49 @@ func main() {
 
 	verboseLog(fmt.Sprintf("Disk info for %s: %s", *disk, info.String()))
 
+	switch {
+	case !info.SecurityStatus.SecurityEnabled():
+		log.Printf("disk security is not enabled on %v", *disk)
+		return
+	case info.SecurityStatus.SecurityFrozen():
+		// If the disk is frozen, its security state cannot be changed until the next
+		// power on or hardware reset. Disk unlock will fail anyways, so return.
+		// This is unlikely to occur, since someone would need to freeze the drive's
+		// security state between the last AC cycle and this code being run.
+		log.Print("disk security is frozen. Power cycle the machine to unfreeze the disk")
+		return
+	case !info.SecurityStatus.SecurityLocked():
+		log.Print("disk is already unlocked")
+		return
+	case info.SecurityStatus.SecurityCountExpired():
+		// If the security count is expired, this means too many attempts have been
+		// made to unlock the disk. Reset this with an AC cycle or hardware reset
+		// on the disk.
+		log.Print("security count expired on disk. Reset the password counter by power cycling the disk.")
+		return
```

Comment:
Should this return an error code?

Reply:
Hm I changed it to log.Fatal to indicate that there was an error, is that ok?